bevy_mod_scripting_core/src/lib.rs

```diff
@@ -75,11 +75,15 @@ pub trait AddScriptHost {
     /// the given stage will contain systems handling script loading,re-loading, removal etc.
     /// This stage will also send events related to the script lifecycle.
     /// Any systems which need to run the same frame a script is loaded must run after this stage.
-    fn add_script_host<T: ScriptHost, S: StageLabel>(&mut self, stage: S) -> &mut Self;
+    fn add_script_host<T: ScriptHost + Resource, S: StageLabel>(&mut self, stage: S) -> &mut Self;
 }
 
 impl AddScriptHost for App {
-    fn add_script_host<T: ScriptHost, S: StageLabel>(&mut self, stage: S) -> &mut Self {
+    fn add_script_host<T, S>(&mut self, stage: S) -> &mut Self
+    where
+        T: ScriptHost + Resource,
```

Comment:
I think we can simply push this `Resource` bound into the ScriptHost definition itself to remove a lot of these explicit restrictions, It'll make everything look cleaner!

Reply:
Did this.

Reply:
Thanks!